app/services/order_detail_updater.rb

```diff
@@ -0,0 +1,34 @@
+module NUCore
+  class QuantityUpdateChangeException < StandardError
+  end
+end
+
+class OrderDetailUpdater
+  attr_reader :order, :params
+
+  def initialize(order, params)
+    @order = order
+    @params = params
+    @initial_quantities = quantities
+  end
+
+  def quantities_changed?
+    @quantities_changed
+  end
+
+  def update
+    update_result = order.update_details(params)
+    @quantities_changed = (@initial_quantities != quantities)
+    update_result
+  end
+
+  def update!
```

Comment:
Do we really need this? I'd love to get rid of the exception handling if we can help it.